### PR TITLE
Add Semigroup instances

### PR DESCRIPTION
**Pull request by fumieval**

### CODE REVIEW

File: enummapset-th.cabal [hunk 1]
```diff
@@ -42,6 +42,7 @@ library
         base >= 4.5 && < 5,
         deepseq >= 1.3,
         containers >= 0.5.3 && < 0.6,
+        semigroups,
```
Comment (pacak) [inline enummapset-th.cabal]: What version of base will include semigroups?

Comment (fumieval) [reply enummapset-th.cabal]: From base-4.9.0.0. Maybe it's fine to drop support for older versions.

Comment (pacak) [reply enummapset-th.cabal]: We should be able to support both older version + semigroups and new version - only base. There seems to be other users as well..

Comment (Mikolaj) [reply enummapset-th.cabal]: The following (based on official docs, can't remember the link) worked for me in https://github.com/Mikolaj/miniutter 

  if !impl(ghc >= 8.0)
    build-depends: semigroups == 0.18.*

and an extra

#if !(MIN_VERSION_base(4,11,0))
  -- this is redundant starting with base-4.11 / GHC 8.4
  mappend = (<>)
#endif

after Semigroup instance is defined. Not tested for this package.

Comment (Mikolaj) [reply enummapset-th.cabal]: Got it: https://prime.haskell.org/wiki/Libraries/Proposals/SemigroupMonoid#Writingcompatiblecode

Comment (Mikolaj) [reply enummapset-th.cabal]: Guys, may I help with this one? I need enummapset-th to work with GHC 8.4.

**Comment by llelf** [comment]
ping!
What ghc versions do you want to support?

**Comment by pacak** [comment]
> ping!
> What ghc versions do you want to support?

Usually it's 3 most major releases but 8.6 is badly broken right now so probably 8.4, 8.2 and 8.0.

**Comment by mitchellwrosen** [comment]
I think it would still make sense to support 8.6, since the fix will not be a breaking change (pvp-wise), so this package would just need to work with base-4.12.

**Comment by pacak** [comment]
Sure, supporting more versions is good, but at least those 3.

**Pull request merged**